src/display/editor/outliner.js

```diff
@@ -582,15 +597,26 @@ class FreeOutliner {
     const lastBottom = last.subarray(16, 18);
     const [layerX, layerY, layerWidth, layerHeight] = this.#box;
 
-    const points = new Float64Array(this.#points?.length ?? 0);
+    const points = new Float64Array((this.#points?.length ?? 0) + 2);
     for (let i = 0, ii = points.length; i < ii; i += 2) {
```

Comment:
Isn't this loop covering more than necessary now?
```suggestion
    for (let i = 0, ii = points.length - 2; i < ii; i += 2) {
```